iOS-DOCS.md

```diff
@@ -0,0 +1,393 @@
+
+It's important to note that the player should not immediately resume playback once receiving the `onAdFreePod` event -- rather, it should note that it was fired and continue to wait for a terminal event.
+
+
+### Handling Ad Elimination
+
+Skipping video ads is completely the responsibility of the app code. The SSAI API should provide enough information for the app to determine where the current pod end-point is, and the app, when appropriate, should fast-forward directly to this point when resuming playback.
+
+
+## TruexAdRenderer iOS API
+
+This is an outline of public `TruexAdRenderer` methods and `TruexAdRendererDelegate` methods (referred to as events).
+
+
+### Adding `TruexAdRenderer` to Your Project
+
+The easiest way to add `TruexAdRenderer` is via CocoaPods. The renderer will be maintained and distributed on Github.
+
+1. Add the following Podspec repositories to your Podfile:
+    ```ruby
+    source 'https://github.com/socialvibe/cocoapod-specs.git'
+    ```
+1. Add the TruexAdRenderer pod to your target:
+    ```ruby
+    target 'MyApp' do
+        pod 'TruexAdRenderer-iOS', '~> 3.2'
+        # [...]
+    end
+    ```
+
+### TruexAdRenderer Methods
+
+#### `initWithUrl`
+
+```objective-c
+    - (_Nullable id)initWithUrl:(NSString* _Nullable)creativeUrl adParameters:(NSDictionary* _Nonnull)adParameters slotType:(NSString* _Nonnull)slotType;
+```
+
+This method should be called by the app code in order to initialize the `TruexAdRenderer`. The renderer will parse out the `creativeURL`, `adParameters` and `slotType` passed to it and make a request to the true[X] ad server to see what ads are available.
+
+You may initialize `TruexAdRenderer` early (a few seconds before the next pod even starts) in order to give it extra time to make the ad request. The renderer will output an [`onFetchAdComplete`](#onfetchadcomplete) event at completion of this ad request. This event can be used to facilitate the implementation of a timeout or loading indicator, and when to make the call to `start`. Alternatively, you may pass in a completion block, which will be triggered when `TruexAdRenderer` is ready.
+
+The parameters for this method call are:
+
+* `creativeURL`: true[X] asset url returned by SSAI. In the example of Uplynk, this would correspond to `response.ads.breaks[0].ads[0].creative`
+* `adParameters`: AdParameters as returned by SSAI. In the example of Uplynk, this would correspond to `response.ads.breaks[0].ads[0].adParameters`
+* `slotType`: the type of the current ad pod, `PREROLL` or `MIDROLL`
+
+
+#### `start`
+
+```objective-c
+    - (void)start:(UIView *)baseView;
+```
+
+This method should be called by the app code when the app is ready to display the true[X] unit to the user. This can be called anytime after the unit is initialized. 
+
+The app should have as much extraneous UI hidden as possible, including player controls, status bars and soft buttons/keyboards.
```

Comment:
should the host app actually unload the main video as well, is that needed on actual devices?
it certainly is a good idea for the HTML devices

Reply:
not require on mobile